distributedlog-core/src/main/java/com/twitter/distributedlog/BKSyncLogReaderDLSN.java

```diff
@@ -116,7 +110,7 @@ public void resumeReadAhead() {
 
     @Override
     public void onSuccess(LogRecordWithDLSN record) {
-        setLastSeenDLSN(record.getDlsn());
+        this.lastSeenDLSN = record.getDlsn();
```

Comment:
Are you referring this code below? lastDLSNSeenByReadAhead is a local variable and the lastSeenDLSN is a volatile variable. The visibility and the ordering of a volatile variable is guaranteed by java - that says accessing the lastSeenDLSN will retrieve the latest value, hence no atomicity is required here.

`                        // if last seen DLSN by reader is same as the one seen by ReadAhead
                        // that means that reader is caught up with ReadAhead and ReadAhead
                        // is caught up with stream
                        shallWait = DLSN.InitialDLSN != lastDLSNSeenByReadAhead
                                && lastSeenDLSN.compareTo(lastDLSNSeenByReadAhead) < 0
                                && startDLSN.compareTo(lastDLSNSeenByReadAhead) <= 0;`